Proposed patch from Debian bug tracker by Aurelien Jarno <aurel32@debian.org>

diff --git a/sh.proc.c b/sh.proc.c
index ad07250..5c68409 100644
--- a/sh.proc.c
+++ b/sh.proc.c
@@ -47,11 +47,11 @@ RCSID("$tcsh$")
 # define HZ 16
 #endif /* aiws */
 
-#if defined(_BSD) || (defined(IRIS4D) && __STDC__) || defined(__lucid) || defined(__linux__) || defined(__GNU__) || defined(__GLIBC__)
-# if !defined(__ANDROID__)
+#if defined(_BSD) || (defined(IRIS4D) && __STDC__) || defined(__lucid) || defined(__linux__) || defined(__GLIBC__)
+# if !defined(__ANDROID__) && !defined(__GLIBC__)
 #  define BSDWAIT
 # endif
-#endif /* _BSD || (IRIS4D && __STDC__) || __lucid || glibc */
+#endif /* _BSD || (IRIS4D && __STDC__) || __lucid || gnu-linux */
 #ifndef WTERMSIG
 # define WTERMSIG(w)	(((union wait *) &(w))->w_termsig)
 # ifndef BSDWAIT
